internal/provider/job_resource_test.go

```diff
@@ -415,6 +417,40 @@ func TestAccAAPJob_UpdateWithTrigger(t *testing.T) {
 	})
 }
 
+// TestAccAAPJob_WaitForCompletion tests that job status is correctly updated to final state
+// when wait_for_completion=true. This test demonstrates the bug described in AAP-47221.
+// Expected to FAIL on main branch, PASS after PR #131 and #132 are merged.
+func TestAccAAPJob_WaitForCompletion(t *testing.T) {
+	jobTemplateID := os.Getenv("AAP_TEST_JOB_FOR_HOST_RETRY_ID")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccJobResourcePreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccJobWithWaitForCompletion(jobTemplateID),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckJobExists,
+					// This check should FAIL on main branch due to AAP-47221 bug
+					// The job status should be "successful" or "failed", not "pending"
+					resource.TestCheckResourceAttrWith("aap_job.test", "status", func(value string) error {
+						if value == statusPendingConst {
+							return fmt.Errorf("AAP-47221 bug: job status is 'pending' instead of final state")
+						}
+						if !IsFinalStateAAPJob(value) {
```

Comment:
Is the above `value == statusPendingConst` necessary if what we really care about is that the job is in a "final" state?